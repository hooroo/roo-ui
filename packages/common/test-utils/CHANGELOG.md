# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
