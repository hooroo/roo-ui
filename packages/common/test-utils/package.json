{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
