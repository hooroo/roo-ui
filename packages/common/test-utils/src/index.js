export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
