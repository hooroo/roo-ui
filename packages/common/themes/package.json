{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
