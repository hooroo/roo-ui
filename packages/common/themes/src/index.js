import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
