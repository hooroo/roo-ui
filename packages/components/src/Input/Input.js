import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import {
  space,
  color,
  fontSize,
  lineHeight,
  border,
  borderColor,
  themeGet,
  variant,
} from 'styled-system';
import tag from 'clean-tag';

const readonlyVariant = variant({
  prop: 'readonlyVariant',
  key: 'readonlyStyles',
});

const Input = styled(tag)`
  ${space}
  ${color}
  ${fontSize}
  ${lineHeight}
  ${border}
  ${borderColor}
  display: block;
  width: 100%;
  outline: 0;
  transition: border-color ${themeGet('transitions.default')};
  appearance: none;

  &:focus {
    border-color: ${themeGet('colors.brand.secondary')};
  }

  &:disabled {
    opacity: ${themeGet('opacity.disabled')};
    cursor: not-allowed;
  }

  &[readonly] {
    ${readonlyVariant}
  }

  ::placeholder {
    color: ${themeGet('colors.greys.steel')};
  }

  ::-ms-clear {
    display: none;
  }

  ${props => props.underline && css`
    border: none;
    border-bottom: ${themeGet('borders.2')};
    border-color: ${themeGet('colors.greys.alto')};
  `}

  ${props => props.error && css`
    border-color: ${themeGet('colors.ui.error')};
  `}
`;

Input.propTypes = {
  ...space.propTypes,
  ...color.propTypes,
  ...fontSize.propTypes,
  ...lineHeight.propTypes,
  ...border.propTypes,
  ...borderColor.propTypes,
  error: PropTypes.bool,
  underline: PropTypes.bool,
  readonlyVariant: PropTypes.string,
};

Input.defaultProps = {
  is: 'input',
  mb: 3,
  py: 3,
  px: 4,
  bg: 'white',
  color: 'greys.charcoal',
  fontSize: 'base',
  lineHeight: 'normal',
  border: 2,
  borderColor: 'greys.alto',
  readonlyVariant: 'naked',
  blacklist: Object.keys(Input.propTypes),
};

export default Input;
