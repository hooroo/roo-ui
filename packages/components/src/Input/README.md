# Input

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Input } from '@roo-ui/components';

export default (
  <Input placeholder="Hello world" />
);
```

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [space](https://github.com/jxnblk/styled-system#space-responsive).
