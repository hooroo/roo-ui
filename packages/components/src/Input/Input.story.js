import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { boolean } from '@storybook/addon-knobs/react';

import Input from '.';
import README from './README.md';

storiesOf('Components|Input', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Input
      placeholder="Hello world"
      error={boolean('Error', false)}
      readOnly={boolean('Read only', false)}
    />
  ));
