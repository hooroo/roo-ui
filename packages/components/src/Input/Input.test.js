import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';
import { axe } from 'jest-axe';

import Input from '.';

describe('<Input />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Input />, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    wrapper = shallowWithTheme(<label htmlFor="input">Label<Input /></label>, theme);
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });

  describe('underline', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<Input underline />, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
