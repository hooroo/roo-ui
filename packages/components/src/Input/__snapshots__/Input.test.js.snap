// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders correctly 1`] = `
.c0 {
  display: block;
  width: 100%;
  margin: 0;
  padding: 0.75rem 1rem;
  font-size: 1rem;
  line-height: 1.35;
  color: #323232;
  border: 2px solid;
  border-color: #DADADA;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c0:focus {
  border-color: #8DE2E0;
}

.c0::-webkit-input-placeholder {
  color: #666666;
}

.c0::-moz-placeholder {
  color: #666666;
}

.c0:-ms-input-placeholder {
  color: #666666;
}

.c0::placeholder {
  color: #666666;
}

.c0::-ms-clear {
  display: none;
}

<Clean.input
  className="c0"
/>
`;
